spoc/pixels.py

```diff
@@ -0,0 +1,326 @@
+"""This part of spoc is responsible for binned,
+higher order contacts in the form of 'genomic pixels'"""
+from pathlib import Path
+import pandas as pd
+import dask.dataframe as dd
+import bioframe as bf
+import pyranges as pr
+from spoc.models import HigherOrderContactSchema
+
+
+class PersistedPixels:
+    """Wrapper for path to pixel parquet file.
+    Responsible for validation. This breaks the IO vs domain logic layer,
+    but improves performance of pileup."""
+
+    def __init__(self, path: str):
+        # check whether path exists
+        if not Path(path).exists():
+            raise ValueError(f"Path: {path} does not exist!")
+        self._path = path
```

Comment:
Maybe wrap it as `self._path = Path(path)`?

Reply:
Sounds good, will do that (after checking whether all downstream clients can handle this)!